frontend/src/helper/CSVConverter.tsx

```diff
@@ -20,7 +36,11 @@ const CSVConverter = (data: LogRecord[]): boolean => {
     const url = URL.createObjectURL(blob);
     const link = document.createElement("a");
     link.href = url;
-    link.setAttribute("download", "log_records.csv");
+    
+    const dateToday = new Date();
+    const dateTodayString = dateToday.toLocaleString("fr-CA", {timeZone: "America/Toronto"}).substring(0, 10);
```

Comment:
nit: Maybe add a comment above explaining why we're leveraging the `fr-CA` formatter 

For instance,

```suggestion
    // Format the date into YYYY-MM-DD 
    const dateToday = new Date();
    const dateTodayString = dateToday.toLocaleString("fr-CA", {timeZone: "America/Toronto"}).substring(0, 10);
```